packages/mongodb-rag-core/src/contentStore/updateEmbeddedContent.ts

```diff
@@ -18,6 +18,21 @@ export interface EmbedConcurrencyOptions {
   createChunks?: number;
 }
 
+const chunkAlgoHashes = new Map<string, string>();
```

Comment:
this is to cache existing hashes so you dont have to recompute? 

also preferable to scope this within a function `updateEmbeddedContent()`. better not to instantiate things at the global scope if you can avoid. this way it's not instantiated til the func is called and its cleaned up after the func's run is over then pass the instantiation to the getHashForFunc() func. pseduocode:

function getHashForFunc(f, o, cache) {
// ...
}

function updateEmbeddedContent(/* args*/) {
  const chunkAlgoHashes = new Map<string, string>();
  // do stuff
  const chunkAlgoHash = getHashForFunc(chunkPage, chunkOptions, chunkAlgoHashes);
}

Reply:
I see your point and can make that change. This was existing code that I simply moved up (kept same scope) because I was calling it before it was defined.